tests/test_urls.py

```diff
@@ -63,3 +63,45 @@ def test_something_else():
 
     result = testdir.runpytest_subprocess()
     assert result.ret == 0
+
+def test_urls_cache_is_cleared_and_new_urls_can_be_assigned(testdir):
+    testdir.makepyfile(myurls="""
+        from django.conf.urls import url
+        from pytest_django_test.compat import patterns
+
+        def fake_view(request):
+            pass
+
+        urlpatterns = patterns('', url(r'first/$', fake_view, name='first'))
+    """)
+
+    testdir.makepyfile(myurls2="""
+        from django.conf.urls import url
+        from pytest_django_test.compat import patterns
+
+        def fake_view(request):
+            pass
+
+        urlpatterns = patterns('', url(r'second/$', fake_view, name='first'))
+    """)
+
+    testdir.makepyfile("""
+        from django.core.urlresolvers import reverse, NoReverseMatch
+        import pytest
+
+        @pytest.mark.urls('myurls')
+        def test_something():
+            reverse('first')
+
+
+        @pytest.mark.urls('myurls2')
+        def test_something_else():
+            with pytest.raises(NoReverseMatch):
+                reverse('first')
```

Comment:
@thedrow 
I am not sure if this test is really what you want to test: the `urlpatterns` from `myurls2` uses `name='first'`, so I would expect that it does not throw `NoReverseMatch` here actually?!